service-commands/scripts/provision-dev-env.sh

```diff
@@ -59,15 +59,17 @@ function setup_python() {
     sudo apt install -y "python$PYTHON_VERSION"
     sudo apt install -y "python$PYTHON_VERSION-dev"
 
-    pip install \
+    python$PYTHON_VERSION -m pip install --upgrade pip
+
+    python$PYTHON_VERSION -m pip install \
```

Comment:
Could you format the pip install to be within line 64's `pip install` in the same format as line 66, so `pip==22.0.4`.

This way, we lock our pip to a specific version (to help alleviate future transient issues when pip auto-upgrades silently).

Nice job on using `pythonX -m pip` which has never caused me issues!

Reply:
Since we're having a bit of issues around `A` from different angles, I'd vote to restrict where these issues are coming from by locking down items that could change along our main path to running `A up`.

If we lock down pip to a specific version, that's one less thing we have to worry about shifting. The concern comes from issues like #2900 where the PalletsProject updated `click` and caused us to make a few changes to bump and lock down version.

Since pip is getting more powerful with dependency management (like what Ricky showed over our call), it'd be good to lock down the version as pip reworks their internal code.

If we're not locking pip to a version within our Docker image I'd vote to do that as well, even if through a different PR, but I think there's more flexibility there since the shipped Docker container will be the same for everybody. In this case, two different people can have different versions of pip depending on when they launched/provisioned their machine. This becomes more of a maintenance issue especially as the team is heavily rotating through boxes at the moment.